superset-frontend/src/pages/Chart/Chart.test.tsx

```diff
@@ -0,0 +1,217 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import thunk from 'redux-thunk';
+import fetchMock from 'fetch-mock';
+import configureStore from 'redux-mock-store';
+import { Link } from 'react-router-dom';
+import {
+  render,
+  waitFor,
+  screen,
+  fireEvent,
+} from 'spec/helpers/testing-library';
+import {
+  getDashboardFormData,
+  getExploreFormData,
+} from 'spec/fixtures/mockExploreFormData';
+import { LocalStorageKeys } from 'src/utils/localStorageHelpers';
+import getFormDataWithExtraFilters from 'src/dashboard/util/charts/getFormDataWithExtraFilters';
+import { URL_PARAMS } from 'src/constants';
+
+import ChartPage from '.';
+
+jest.mock('src/explore/components/ExploreViewContainer', () => () => (
+  <div data-test="mock-explore-view-container" />
+));
+jest.mock('src/dashboard/util/charts/getFormDataWithExtraFilters');
+
+const mockStore = configureStore([thunk]);
+
+describe('ChartPage', () => {
+  afterEach(() => {
+    fetchMock.reset();
+  });
+
+  test('fetches metadata on mount', async () => {
+    const store = mockStore({
+      explore: {},
+    });
+    const exploreApiRoute = 'glob:*/api/v1/explore/*';
+    const exploreFormData = getExploreFormData({
+      viz_type: 'table',
+      show_cell_bars: true,
+    });
+    fetchMock.get(exploreApiRoute, {
+      result: { dataset: { id: 1 }, form_data: exploreFormData },
+    });
+    const { getByTestId } = render(<ChartPage />, {
+      useRouter: true,
+      useRedux: true,
+      store,
+    });
+    await waitFor(() =>
+      expect(fetchMock.calls(exploreApiRoute).length).toBe(1),
+    );
+    expect(getByTestId('mock-explore-view-container')).toBeInTheDocument();
+    expect(store.getActions()).toContainEqual({
```

Comment:
@justinpark React Testing Library has a different view on how to write tests. One of its main concerns is accessibility. It differs from Enzyme in the sense that it's more focused on how the user interacts with the app rather then implementation details. Check this quote from their [site](https://testing-library.com/docs/#the-problem):

> You want to write maintainable tests that give you high confidence that your components are working for your users. As a part of this goal, you want your tests to avoid including implementation details so refactors of your components (changes to implementation but not functionality) don't break your tests and slow you and your team down.

In your case, you could write tests that check if the "Show cell bars" is checked rather then test if an specific Redux action is fired. This will make your tests more durable in a scenario where the implementation details change, such as the redux action name, or the action structure, but the interface, and the way users interact with it, remain the same.

Reply:
@michael-s-molina I updated the test code as you suggested. (but using partial rendering since full ChartContainer requires a lot of dependancies)